### PR TITLE
Replace Chakra UI with shadcn-ui

### PR DESCRIPTION
**Pull request by leonardoarcari**
## Description

Chakra UI with the v3 version has broken everything and after a few hours of swearing I discovered that it is incompatible with TailwindCSS. Furthermore, the documentation is still particularly poor. Now, since this is a side-project, thinking of rewriting every component in Chakra just to be able to migrate to the new stable is out of the question. Given that we will have at most 3 components migrated to Chakra, I would be in favor of saying goodbye to the framework.

Among the libraries shared by @corradopetrelli, there is one that seems to be right for us: https://ui.shadcn.com/

In short, it is based on Radix UI (very big project) but integrated with TailwindCSS. The idea is to offer a ton of unstyled components and then with tailwind you customize them. There is still the additional plus that the components are generated within the repo and committed, so no npm package. By doing so we can customize them as we like.

## Demo

https://github.com/user-attachments/assets/f83720bf-3479-4327-a15c-8fa56b2acf05

### CODE REVIEW

File: dcapal-frontend/src/components/languageSwitcher.js [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
+    <Select
+      defaultValue={i18n.resolvedLanguage}
+      onValueChange={(newLang) => {
         i18n.changeLanguage(newLang);
       }}
     >
-      <option value="en"> {t("language.en")} </option>
-      <option value="it"> {t("language.it")} </option>
-    </select>
+      <SelectTrigger className="w-32" aria-label="Change language">
```
Comment (mattiapitossi) [inline dcapal-frontend/src/components/languageSwitcher.js:21]: I'm not too convinced about the size of the language selector button, maybe we can make it smaller?

```suggestion
      <SelectTrigger
        className="h-6 min-w-[4rem] w-auto px-2"
        aria-label="Change language"
      >
```

File: dcapal-frontend/src/components/languageSwitcher.js [hunk 2]
```diff
@@ -1,22 +1,31 @@
...
+        <SelectValue />
+      </SelectTrigger>
+      <SelectContent>
+        <SelectItem value="en">{t("language.en")}</SelectItem>
+        <SelectItem value="it">{t("language.it")}</SelectItem>
+      </SelectContent>
+    </Select>
```
Comment (mattiapitossi) [inline dcapal-frontend/src/components/languageSwitcher.js:28]: When the selector enters in focus mode, the navbar creates a weird shifting effect, I tried to provide a solution without any result. Do you think we can fix it?

Comment (leonardoarcari) [reply dcapal-frontend/src/components/languageSwitcher.js:31]: Found the workaround! It was a known issue: https://github.com/shadcn-ui/ui/issues/4227

File: dcapal-frontend/src/components/languageSwitcher.js (hunk 1 above)
Comment (leonardoarcari) [reply dcapal-frontend/src/components/languageSwitcher.js]: Agree, it was definitely too big. Chose to keep the default height and padding to keep visual consistency as we'll use more of them.

File: dcapal-frontend/src/style.css
Comment (KristianLentino99) [inline dcapal-frontend/src/style.css:1]: What do you think about moving this file into a `styles` folder or similar?

File: dcapal-frontend/src/lib/utils.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { clsx } from "clsx";
+import { twMerge } from "tailwind-merge";
+
+export function cn(...inputs) {
```
Comment (KristianLentino99) [inline dcapal-frontend/src/lib/utils.js:4]: `cn` is not very self-explanatory, is there a better name?

Comment (KristianLentino99) [reply dcapal-frontend/src/lib/utils.js:4]: I think is the default name that you find around when doing this kind of thing, but I tend to prefer self-explanatory names for methods

File: dcapal-frontend/src/style.css
Comment (leonardoarcari) [reply dcapal-frontend/src/style.css:1]: Not sure, it looks like global Tailwind CSS is usually stored in the `src` or `app` folder root

File: dcapal-frontend/src/lib/utils.js (hunk 3 above)
Comment (leonardoarcari) [reply dcapal-frontend/src/lib/utils.js:4]: It's required by `shadcn-ui`. Please take a look at the installation docs: https://ui.shadcn.com/docs/installation/manual

File: dcapal-frontend/src/style.css
Comment (KristianLentino99) [reply dcapal-frontend/src/style.css:1]: ok make sense

File: dcapal-frontend/src/lib/utils.js (hunk 3 above)
Comment (KristianLentino99) [reply dcapal-frontend/src/lib/utils.js:4]: No I know, but I dont think that `cn` name is required. I mean, you can call it however you prefer. The docs is saying that you need to have an utility like that, not how to name it 🤔

Comment (leonardoarcari) [reply dcapal-frontend/src/lib/utils.js:4]: It's mandatory. Take a look at the components generated by the shadcn CLI (`@/components/ui`). You see that they expect a `cn` function to be available.

Comment (KristianLentino99) [reply dcapal-frontend/src/lib/utils.js:4]: Ah ok thanks for the clarification, didnt noticed that

**Review by KristianLentino99 — APPROVED** [review]

**Review by mattiapitossi — APPROVED** [review]
